src/App.tsx

```diff
@@ -1,26 +1,50 @@
+/* eslint-disable max-len */
+import React, { useEffect } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-import { Loader, TodoFilter, TodoList, TodoModal } from './components';
 
-export const App = () => (
-  <>
-    <div className="section">
-      <div className="container">
-        <div className="box">
-          <h1 className="title">Todos:</h1>
+import { TodoList } from './components/TodoList';
+import { TodoFilter } from './components/TodoFilter';
+import { TodoModal } from './components/TodoModal';
+import { Loader } from './components/Loader';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { getTodos } from './api';
+import { actions } from './features/todos';
 
-          <div className="block">
-            <TodoFilter />
-          </div>
+export const App: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const hasTodos = useAppSelector(state => state.todos.length > 0);
+  const hasSelectedTodo = useAppSelector(state => state.currentTodo !== null);
+
+  useEffect(() => {
+    getTodos().then(todos => dispatch(actions.setTodos(todos)));
+  });
```

Comment:
The `useEffect` hook is missing a dependency array. This can cause the effect to run on every render, leading to potential performance issues or infinite loops. Consider adding an empty dependency array `[]` to ensure the effect runs only once when the component mounts.